### PR TITLE
MH-13424, Elasticsearch 5.6.15

### PR DESCRIPTION
**Pull request by lkiesow**
This patch updates Opencast's Elasticsearch compatibility and the
internal Elasticsearch node to version 5.6.15, finally btinging us to a
recent (Feb 19, 2019), still supported version.

### CODE REVIEW

**Comment by miesgre** [comment]
@lkiesow Why elasticsearch 5 and not elasticsearch 6?
https://www.elastic.co/support/eol

**Comment by lkiesow** [comment]
Elasticsearch 6 is still planned but it may not be feasible to integrate it in the same manner.
It also does not make sense to wait until I finish the 6.x upgrade when 5 is already ready to go.
As announced earlier, the plan is to go 1.6.x → 2.4.x → 5.6.x → *6.x*.
There are enough changes already when going these small steps.

File: modules/search/pom.xml [hunk 1]
```diff
@@ -39,9 +39,44 @@
...
       <groupId>org.elasticsearch</groupId>
       <artifactId>elasticsearch</artifactId>
-      <version>2.4.6</version>
+      <version>5.6.15</version>
     </dependency>
-    <!-- elasticsearch dependencies -->
+    <dependency>
+      <groupId>org.elasticsearch.client</groupId>
+      <artifactId>transport</artifactId>
+      <version>5.6.15</version>
```
Comment (staubesv) [inline modules/search/pom.xml]: Is there a special reason for listing the version many times vs. using a variable like ${elasticsearch.version} similar to ${lucene.version}?

**Review by staubesv — DISMISSED** [review]
Just the question about using a variable for the Elasticsearch version, besides that looks good to me.

**Review by staubesv — APPROVED** [review]

**Pull request merged**